guides/04_additional-features/08_file-access.md

```diff
@@ -0,0 +1,64 @@
+# File Access
+
+Sharing your Gradio app with others (by hosting it on Spaces, on your own server, or through temporary share links) **exposes** certain files on your machine to the internet.
+
+This guide will explain which ones as well as some best practices for making sure the files on your machine are secure.
+
+## The Gradio cache
+
+First, it's important to understand that Gradio places files in a special `cache` before returning them to the frontend. For example, if your prediction function returns a video file, then Gradio will move that video to the `cache` after your prediction function runs and returns a URL the frontend can use to show the video. Any file in the `cache` is available via URL while the application is running.
+
+Tip: You can customize the location of the `cache` by setting the `GRADIO_TEMP_DIR` environment variable to an absolute path, such as `/home/usr/scripts/project/temp/`. 
+
+## The files Gradio will move to the cache
+
+Before placing a file in the cache, Gradio will check to see if the file meets at least one of following criteria:
+
+1. It was uploaded by a user.
+2. It is in the `allowed_paths` parameter of the `Blocks.launch` method.
+3. It is in the current working directory of the python interpreter.
+4. It is in the temp directory obtained by `tempfile.gettempdir()`.
+
+Additionally, files in the current working directory whose name starts with a period (`.`) will not be moved to the cache. If no criteria are met, the prediction function that created that file will error. Gradio performs this check so that arbitrary files on your machine are not moved to the cache.
```

Comment:
```suggestion
Note: files in the current working directory whose name starts with a period (`.`) will not be moved to the cache, since they often contain sensitive information for your application. 

If none of these criteria are met, the prediction function that created that file will raise an exception instead of moving the file to cache. Gradio performs this check so that arbitrary files on your machine cannot be accessed.
```